FirebasePerformance/Sources/AppActivity/FPRAppActivityTracker.m

```diff
@@ -143,24 +146,49 @@ - (FIRTrace *)activeTrace {
 }
 
 - (BOOL)isApplicationPreWarmed {
-  NSArray *versionComponents =
-      [[UIDevice currentDevice].systemVersion componentsSeparatedByString:@"."];
-  NSInteger majorVersion = [versionComponents[0] integerValue];
-  if (majorVersion < 15) {
+  if (![self prewarmAvailable]) {
     return NO;
   }
 
-  NSDictionary<NSString *, NSString *> *environment = [NSProcessInfo processInfo].environment;
-  if (environment[@"ActivePrewarm"] != nil &&
-      [environment[@"ActivePrewarm"] isEqualToString:@"1"]) {
-    return YES;
-  }
+  return [self isActivePrewarm] || [self isDoubleDispatchPrewarm];
+}
 
-  if ([doubleDispatchTime compare:applicationFinishLaunchTime] == NSOrderedAscending) {
-    return YES;
+- (void)applyPrewarmTag:(FIRTrace *)trace {
+  BOOL activePrewarm = [self isActivePrewarm];
+  BOOL doubleDispatchPrewarm = [self isDoubleDispatchPrewarm];
+  if (![self prewarmAvailable]) {
+    return;
+  }
+  if (!activePrewarm && !doubleDispatchPrewarm) {
+    [trace setValue:@"cold" forAttribute:@"prewarm_detection"];
+    [trace setValue:@"no" forAttribute:@"is_prewarmed"];
+  } else if (activePrewarm && doubleDispatchPrewarm) {
+    [trace setValue:@"both" forAttribute:@"prewarm_detection"];
+    [trace setValue:@"yes" forAttribute:@"is_prewarmed"];
+  } else if (activePrewarm && !doubleDispatchPrewarm) {
+    [trace setValue:@"active_prewarm" forAttribute:@"prewarm_detection"];
+    [trace setValue:@"yes" forAttribute:@"is_prewarmed"];
+  } else {
+    [trace setValue:@"double_dispatch" forAttribute:@"prewarm_detection"];
+    [trace setValue:@"yes" forAttribute:@"is_prewarmed"];
   }
+}
+
+- (BOOL)prewarmAvailable {
+  NSArray *versionComponents =
+      [[UIDevice currentDevice].systemVersion componentsSeparatedByString:@"."];
+  NSInteger majorVersion = [versionComponents[0] integerValue];
+  return majorVersion >= 15;
+}
+
+- (BOOL)isActivePrewarm {
+  NSDictionary<NSString *, NSString *> *environment = [NSProcessInfo processInfo].environment;
+  return (environment[@"ActivePrewarm"] != nil &&
```

Comment:
May want to use a plist value to check for enabling the EAP code to make it easy to include in a release and avoid separate release management for the EAP.  For example, see https://github.com/firebase/firebase-ios-sdk/issues/8949#issuecomment-975970983

Reply:
Do you mean we make a normal release, then ask EAP customers to set a plist value to enable this prototype?

Reply:
Yes. It's easier from a documentation/release management/installation/usability perspective.

Reply:
Discussed offline and agreed this is appropriate for a limited EAP